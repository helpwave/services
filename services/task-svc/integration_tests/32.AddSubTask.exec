#!/bin/bash


echo "no optionals"
id=$(grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"name": "Some Great Task", "task_id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c"}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/AddSubTask | jq -r ".id")

echo "result:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c"}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask | jq '.subtasks[] |= . + { "id": "RANDOM ID" }'

echo "delete again"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/RemoveSubTask

echo "see:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c"}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask

echo "with optionals"
id=$(grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"name": "Some Great Task", "task_id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c", "done": true}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/AddSubTask | jq -r ".id")

echo "result:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c"}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask | jq '.subtasks[] |= . + { "id": "RANDOM ID" }'

echo "delete again"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/RemoveSubTask

echo "see:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c"}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask
