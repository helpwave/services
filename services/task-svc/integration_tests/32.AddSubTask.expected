no optionals
result:
{
  "id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c",
  "name": "Changed Name",
  "description": "Clear description",
  "status": "TASK_STATUS_DONE",
  "assignedUserId": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758",
  "patient": {
    "id": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
    "name": "Grazia Rehn"
  },
  "public": true,
  "dueAt": "2023-11-04T06:58:00Z",
  "createdBy": "df363387-7774-4592-b01c-2b765abaa86a",
  "subtasks": [
    {
      "id": "bf154d1e-19d7-4d62-9864-cf34805b91fe",
      "name": "Some Great Task",
      "createdBy": "18159713-5d4e-4ad5-94ad-fbb6bb147984"
    }
  ],
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
delete again
{}
see:
{
  "id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c",
  "name": "Changed Name",
  "description": "Clear description",
  "status": "TASK_STATUS_DONE",
  "assignedUserId": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758",
  "patient": {
    "id": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
    "name": "Grazia Rehn"
  },
  "public": true,
  "dueAt": "2023-11-04T06:58:00Z",
  "createdBy": "df363387-7774-4592-b01c-2b765abaa86a",
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
with optionals
result:
{
  "id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c",
  "name": "Changed Name",
  "description": "Clear description",
  "status": "TASK_STATUS_DONE",
  "assignedUserId": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758",
  "patient": {
    "id": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
    "name": "Grazia Rehn"
  },
  "public": true,
  "dueAt": "2023-11-04T06:58:00Z",
  "createdBy": "df363387-7774-4592-b01c-2b765abaa86a",
  "subtasks": [
    {
      "id": "4cef3c3a-2ee9-4529-8f09-1509ca321c18",
      "name": "Some Great Task",
      "done": true,
      "createdBy": "18159713-5d4e-4ad5-94ad-fbb6bb147984"
    }
  ],
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
delete again
{}
see:
{
  "id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c",
  "name": "Changed Name",
  "description": "Clear description",
  "status": "TASK_STATUS_DONE",
  "assignedUserId": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758",
  "patient": {
    "id": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
    "name": "Grazia Rehn"
  },
  "public": true,
  "dueAt": "2023-11-04T06:58:00Z",
  "createdBy": "df363387-7774-4592-b01c-2b765abaa86a",
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
